en/docs/extended_topics/python_kernel_op.md

```diff
@@ -0,0 +1,291 @@
+# Extending Op with Python
+**Attention** : The Python Kernel covered in this article was only fully tested under the `gcc 4.8.5` compilation environment. Further refinements are planned in [Issue 3951](https://github.com/Oneflow-Inc/oneflow/issues/3951).
+
+## Background Information
+OneFlow abstracts all kinds of data processing into operators. Op is act on the input tensor and writes the result of the calculation to the output tensor. A comprehensive operator is available inside OneFlow which can be found in the [ops directory](https://github.com/Oneflow-Inc/oneflow/tree/master/oneflow/python/ops).
+
+When OneFlow's existing Python operators and their combinations are not sufficient for building neural networks or when the Python level operators do not meet performance requirements. Then we can develop custom ops. OneFlow provides two ways to develop custom Ops. One is the Python based `Python Kernel` development and the other is the `C++ Kernel` development which is introduced in the article [Extending Op with C++](./user_op.md)
+
+`Python Kernel` has a simple development process and is suitable for rapid pre-research, algorithm verification and other scenarios because it is mainly extended by Python. The `C++ Kernel` is efficient and suitable for developing operators that have proven reliability and are pursue for performance .
```

Comment:
```suggestion
`Python Kernel` has a simple development process and is suitable for pre-research, algorithm verification and other scenarios because it is mainly extended by Python. The `C++ Kernel` is efficient and suitable for developing operators that have been proven reliability and with performance requirement.
```